test/functional/provider/passwords_controller_test.rb

```diff
@@ -41,4 +41,11 @@ def setup
     assert_redirected_to back
   end
 
+  test '#destroy also works for master account' do
```

Comment:
The functional tests (inheriting from `ActionController::TestCase`) are deprecated and we definitely should not add new ones. This test should be inside this other class:
https://github.com/3scale/porta/blob/185db8beaffcfb6ea6db5334d045b2bcb00525f6/test/integration/provider/passwords_test.rb#L3

And should look more or less like this:
```ruby
test '#destroy also works for master account' do
  user = FactoryBot.create(:admin, account: master_account)
  login! master_account

  delete provider_password_path(email: user.email)
  assert_response :found
  assert_match 'A password reset link has been emailed to you.', flash[:notice]
end
```